query-node/mappings/src/membership.ts

```diff
@@ -651,46 +653,62 @@ async function processMemberRemark(
   store: DatabaseManager,
   event: SubstrateEvent,
   memberId: MemberId,
-  decodedMessage: MemberRemarked
+  decodedMetadata: DecodedMetadataObject<IMemberRemarked> | null
 ): Promise<Partial<MetaprotocolTransactionSuccessful>> {
-  const messageType = decodedMessage.memberRemarked
+  if (decodedMetadata?.createApp) {
+    await processCreateAppMessage(store, event, decodedMetadata.createApp!, memberId.toString())
 
-  if (messageType === 'reactVideo') {
-    await processReactVideoMessage(store, event, memberId, decodedMessage.reactVideo!)
+    return {}
+  }
+
+  if (decodedMetadata?.updateApp) {
+    await processUpdateAppMessage(store, event, decodedMetadata.updateApp!, memberId.toString())
+
+    return {}
+  }
+
+  if (decodedMetadata?.deleteApp) {
+    await processDeleteAppMessage(store, event, decodedMetadata.deleteApp!, memberId.toString())
+
+    return {}
+  }
+
+  if (decodedMetadata?.reactVideo) {
+    await processReactVideoMessage(store, event, memberId, decodedMetadata.reactVideo!)
 
     return {}
   }
 
-  if (messageType === 'reactComment') {
-    await processReactCommentMessage(store, event, memberId, decodedMessage.reactComment!)
+  if (decodedMetadata?.reactComment) {
+    await processReactCommentMessage(store, event, memberId, decodedMetadata.reactComment!)
 
     return {}
   }
 
-  if (messageType === 'createComment') {
-    const comment = await processCreateCommentMessage(store, event, memberId, decodedMessage.createComment!)
+  if (decodedMetadata?.createComment) {
+    const comment = await processCreateCommentMessage(store, event, memberId, decodedMetadata.createComment!)
 
     return { commentCreatedId: comment.id }
   }
 
-  if (messageType === 'editComment') {
-    const comment = await processEditCommentMessage(store, event, memberId, decodedMessage.editComment!)
+  if (decodedMetadata?.editComment) {
+    const comment = await processEditCommentMessage(store, event, memberId, decodedMetadata.editComment!)
     return { commentEditedId: comment.id }
   }
 
-  if (messageType === 'deleteComment') {
-    const comment = await processDeleteCommentMessage(store, event, memberId, decodedMessage.deleteComment!)
+  if (decodedMetadata?.deleteComment) {
+    const comment = await processDeleteCommentMessage(store, event, memberId, decodedMetadata.deleteComment!)
```

Comment:
Inside this if-block the non-null assertion is no longer needed:
```suggestion
    const comment = await processDeleteCommentMessage(store, event, memberId, decodedMetadata.deleteComment)
```